### ISSUE TITLE
Cant use return?

### ISSUE THREAD

**Issue by 22DP3BBiks** [FrederoxDev/CC-Bedrock#3]
What does exist, do you have like file with the lang gammer?

![image](https://github.com/FrederoxDev/CC-Bedrock/assets/103173909/b1ff0376-32e6-4640-88a9-d5eebf5ec4ff)

**Comment by FrederoxDev** [comment]
Like I stated in previous issues, this is an uncompleted language. The error is stating that the returnExpression has not been implemented in the interpreter

I left it in the middle of a re-write, I don't remember what branch of Cosmic it is on. But there is an older version which I believe had more progress compared to the newer branch

**Comment by FrederoxDev** [comment]
https://github.com/FrederoxDev/Cosmic/blob/master/src/Interpreter.ts

**Comment by 22DP3BBiks** [comment]
Thanks, guess this will require a lot more writing. By the way, how long did it take you to write that interpreter?

**Comment by FrederoxDev** [comment]
It was a while ago, I really don't remember. But if you plan on touching Cosmic, I would advise working on the new branch. It had a lot more of a capable lexer and parser, it's only a bit behind on its interpreter

**Comment by FrederoxDev** [comment]
I might actually have a bit of free time to work on Cosmic soon. No promises tho

**Comment by 22DP3BBiks** [comment]
Yeah, i noticed in the cosmic new breach there is a gammer.txt for grammar, but because the addon using the master breach not the new one, that was the reason for  asking what was the grammer.

Well, i was think of maybe trying to write a some sort of interface were you can connect the language. That way having the ability to swap it, maybe with the same, but different version or completely different.

I could try fixing it, but i would probably spend week just trying to understand how it works.

**Comment by FrederoxDev** [comment]
I recon that the grammar.txt file in the new branch was just an example of grammar and not what I was trying to achieve

**Comment by 22DP3BBiks** [comment]
But what do you think about that interface idea?

**Comment by FrederoxDev** [comment]
I don't ever plan on any other language support myself apart from the new cosmic branch, so I don't really know if it'd have a point for my own use case

**Comment by 22DP3BBiks** [comment]
Okey

**Comment by 22DP3BBiks** [comment]
Kind of a problem that i have no idea what you are planning

**Comment by FrederoxDev** [comment]
Right now in the short term, I want to implement all of these nodes on the new Interpreter which I will start some work on today. https://github.com/FrederoxDev/Cosmic/blob/CosmicRewrite/src/Parser.ts#L10-#L18 

Right now I need to catch up myself where the project got to

**Comment by 22DP3BBiks** [comment]
comment or todo list would help, but whom am i to say that

**Comment by FrederoxDev** [comment]
I'll write one up after I re-orient myself first

**Comment by FrederoxDev** [comment]
The language was actually a tad further than I thought, those statements were seemingly already implemented. Do you have discord? It may be easier to communicate there

**Comment by 22DP3BBiks** [comment]
Yes, i send you hello in dm
